File: binding/src/libqdebug.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qdebug.html
pub const qdebug = struct {

    /// New constructs a new QDebug object.
    pub fn New(device: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_new(@ptrCast(device));
    }


    /// New2 constructs a new QDebug object.
    pub fn New2(stringVal: []const u8) QtC.QDebug {
        const stringVal_str = qtc.libqt_string{
    .len = stringVal.len,
    .data = stringVal.ptr,
};
return qtc.QDebug_new2(stringVal_str);
    }


    /// New3 constructs a new QDebug object.
    pub fn New3(o: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_new3(@ptrCast(o));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDebug_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QDebug_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn ResetFormat(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_ResetFormat(@ptrCast(self));
    }

    pub fn Space(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_Space(@ptrCast(self));
    }

    pub fn Nospace(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_Nospace(@ptrCast(self));
    }

    pub fn MaybeSpace(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_MaybeSpace(@ptrCast(self));
    }

    pub fn Verbosity(self: ?*anyopaque, verbosityLevel: i32) QtC.QDebug {
        return qtc.QDebug_Verbosity(@ptrCast(self), verbosityLevel);
    }

    pub fn Verbosity2(self: ?*anyopaque, ) i32 {
        return qtc.QDebug_Verbosity2(@ptrCast(self));
    }

    pub fn SetVerbosity(self: ?*anyopaque, verbosityLevel: i32) void {
        qtc.QDebug_SetVerbosity(@ptrCast(self), verbosityLevel);
    }

    pub fn AutoInsertSpaces(self: ?*anyopaque, ) bool {
        return qtc.QDebug_AutoInsertSpaces(@ptrCast(self));
    }

    pub fn SetAutoInsertSpaces(self: ?*anyopaque, b: bool) void {
        qtc.QDebug_SetAutoInsertSpaces(@ptrCast(self), b);
    }

    pub fn QuoteStrings(self: ?*anyopaque, ) bool {
        return qtc.QDebug_QuoteStrings(@ptrCast(self));
    }

    pub fn SetQuoteStrings(self: ?*anyopaque, b: bool) void {
        qtc.QDebug_SetQuoteStrings(@ptrCast(self), b);
    }

    pub fn Quote(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_Quote(@ptrCast(self));
    }

    pub fn Noquote(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_Noquote(@ptrCast(self));
    }

    pub fn MaybeQuote(self: ?*anyopaque, ) QtC.QDebug {
        return qtc.QDebug_MaybeQuote(@ptrCast(self));
    }

    pub fn OperatorShiftLeft(self: ?*anyopaque, t: QtC.QChar) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft(@ptrCast(self), @ptrCast(t));
    }

    pub fn OperatorShiftLeft2(self: ?*anyopaque, t: bool) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft2(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft3(self: ?*anyopaque, t: i8) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft3(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft4(self: ?*anyopaque, t: i16) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft4(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft5(self: ?*anyopaque, t: u16) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft5(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft8(self: ?*anyopaque, t: i32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft8(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft9(self: ?*anyopaque, t: u32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft9(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft10(self: ?*anyopaque, t: long) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft10(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft11(self: ?*anyopaque, t: unsigned long) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft11(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft12(self: ?*anyopaque, t: i64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft12(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft13(self: ?*anyopaque, t: u64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft13(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft15(self: ?*anyopaque, t: f32) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft15(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft16(self: ?*anyopaque, t: f64) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft16(@ptrCast(self), t);
    }

    pub fn OperatorShiftLeft17(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft17(@ptrCast(self), @ptrCast(t));
    }

    pub fn OperatorShiftLeft19(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        const t_str = qtc.libqt_string{
    .len = t.len,
    .data = t.ptr,
};
return qtc.QDebug_OperatorShiftLeft19(@ptrCast(self), t_str);
    }

    pub fn OperatorShiftLeft23(self: ?*anyopaque, t: []u8) QtC.QDebug {
        const t_str = qtc.libqt_string{
    .len = t.len,
    .data = t.ptr,
};
return qtc.QDebug_OperatorShiftLeft23(@ptrCast(self), t_str);
    }

    pub fn OperatorShiftLeft24(self: ?*anyopaque, t: []const u8) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft24(@ptrCast(self), @ptrCast(t));
    }

    pub fn OperatorShiftLeft25(self: ?*anyopaque, t: ?*anyopaque) QtC.QDebug {
        return qtc.QDebug_OperatorShiftLeft25(@ptrCast(self), @ptrCast(t));
    }

    pub fn MaybeQuote1(self: ?*anyopaque, c: i8) QtC.QDebug {
        return qtc.QDebug_MaybeQuote1(@ptrCast(self), c);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDebug_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qdebugstatesaver.html
pub const qdebugstatesaver = struct {

    /// New constructs a new QDebugStateSaver object.
    pub fn New(dbg: ?*anyopaque) QtC.QDebugStateSaver {
        return qtc.QDebugStateSaver_new(@ptrCast(dbg));
    }


    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QDebugStateSaver_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qnodebug.html
pub const qnodebug = struct {

    /// New constructs a new QNoDebug object.
    pub fn New(other: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_new(@ptrCast(other));
    }


    /// New2 constructs a new QNoDebug object.
    pub fn New2(other: ?*anyopaque) QtC.QNoDebug {
        return qtc.QNoDebug_new2(@ptrCast(other));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNoDebug_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNoDebug_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Space(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_Space(@ptrCast(self));
    }

    pub fn Nospace(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_Nospace(@ptrCast(self));
    }

    pub fn MaybeSpace(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeSpace(@ptrCast(self));
    }

    pub fn Quote(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_Quote(@ptrCast(self));
    }

    pub fn Noquote(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_Noquote(@ptrCast(self));
    }

    pub fn MaybeQuote(self: ?*anyopaque, ) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeQuote(@ptrCast(self));
    }

    pub fn Verbosity(self: ?*anyopaque, param1: i32) QtC.QNoDebug {
        return qtc.QNoDebug_Verbosity(@ptrCast(self), param1);
    }

    pub fn MaybeQuote1(self: ?*anyopaque, param1: i8) QtC.QNoDebug {
        return qtc.QNoDebug_MaybeQuote1(@ptrCast(self), param1);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNoDebug_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/debug.html#types
pub const enums = struct {
    pub const VerbosityLevel = enum {
        pub const MinimumVerbosity: i32 = 0;
        pub const DefaultVerbosity: i32 = 2;
        pub const MaximumVerbosity: i32 = 7;
    };

};

File: binding/src/libqdebug.cpp
#include <QByteArray>
#include <QByteArrayView>
#include <QChar>
#include <QDebug>
#include <QDebugStateSaver>
#include <QIODevice>
#include <QIODeviceBase>
#include <QNoDebug>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qdebug.h>
#include "libqdebug.h"
#include "libqdebug.hxx"

QDebug* QDebug_new(QIODevice* device) {
	 return new QDebug(device);
}

QDebug* QDebug_new2(libqt_string stringVal) {
	 return new QDebug(QString::fromUtf8(stringVal.data, stringVal.len));
}

QDebug* QDebug_new3(const QDebug* o) {
	 return new QDebug(*o);
}

void QDebug_OperatorAssign(QDebug* self, const QDebug* other) {
	self->operator=(*other);
}

void QDebug_Swap(QDebug* self, QDebug* other) {
	self->swap(*other);
}

QDebug* QDebug_ResetFormat(QDebug* self) {
	return new QDebug(self->resetFormat());
}

QDebug* QDebug_Space(QDebug* self) {
	return new QDebug(self->space());
}

QDebug* QDebug_Nospace(QDebug* self) {
	return new QDebug(self->nospace());
}

QDebug* QDebug_MaybeSpace(QDebug* self) {
	return new QDebug(self->maybeSpace());
}

QDebug* QDebug_Verbosity(QDebug* self, int verbosityLevel) {
	return new QDebug(self->verbosity(verbosityLevel));
}

int QDebug_Verbosity2(const QDebug* self) {
	return self->verbosity();
}

void QDebug_SetVerbosity(QDebug* self, int verbosityLevel) {
	self->setVerbosity(verbosityLevel);
}

bool QDebug_AutoInsertSpaces(const QDebug* self) {
	return self->autoInsertSpaces();
}

void QDebug_SetAutoInsertSpaces(QDebug* self, bool b) {
	self->setAutoInsertSpaces(b);
}

bool QDebug_QuoteStrings(const QDebug* self) {
	return self->quoteStrings();
}

void QDebug_SetQuoteStrings(QDebug* self, bool b) {
	self->setQuoteStrings(b);
}

QDebug* QDebug_Quote(QDebug* self) {
	return new QDebug(self->quote());
}

QDebug* QDebug_Noquote(QDebug* self) {
	return new QDebug(self->noquote());
}

QDebug* QDebug_MaybeQuote(QDebug* self) {
	return new QDebug(self->maybeQuote());
}

QDebug* QDebug_OperatorShiftLeft(QDebug* self, QChar* t) {
	return new QDebug(self->operator<<(*t));
}

QDebug* QDebug_OperatorShiftLeft2(QDebug* self, bool t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft3(QDebug* self, char t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft4(QDebug* self, int16_t t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft5(QDebug* self, uint16_t t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft8(QDebug* self, int t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft9(QDebug* self, unsigned int t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft10(QDebug* self, long t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft11(QDebug* self, unsigned long t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft12(QDebug* self, long long t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft13(QDebug* self, unsigned long long t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft15(QDebug* self, float t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft16(QDebug* self, double t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft17(QDebug* self, const char* t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_OperatorShiftLeft19(QDebug* self, const libqt_string t) {
	return new QDebug(self->operator<<(QString::fromUtf8(t.data, t.len)));
}

QDebug* QDebug_OperatorShiftLeft23(QDebug* self, const libqt_string t) {
	return new QDebug(self->operator<<(QByteArray(t.data, t.len)));
}

QDebug* QDebug_OperatorShiftLeft24(QDebug* self, QByteArrayView* t) {
	return new QDebug(self->operator<<(*t));
}

QDebug* QDebug_OperatorShiftLeft25(QDebug* self, const void* t) {
	return new QDebug(self->operator<<(t));
}

QDebug* QDebug_MaybeQuote1(QDebug* self, char c) {
	return new QDebug(self->maybeQuote(c));
}

void QDebug_Delete(QDebug* self) {
    delete self;
}

QDebugStateSaver* QDebugStateSaver_new(QDebug* dbg) {
	 return new QDebugStateSaver(*dbg);
}

void QDebugStateSaver_Delete(QDebugStateSaver* self) {
    delete self;
}

QNoDebug* QNoDebug_new(const QNoDebug* other) {
	 return new QNoDebug(*other);
}

QNoDebug* QNoDebug_new2(QNoDebug* other) {
	 return new QNoDebug(*other);
}

void QNoDebug_CopyAssign(QNoDebug* self, QNoDebug* other) {
    *self = *other;
}

void QNoDebug_MoveAssign(QNoDebug* self, QNoDebug* other) {
    *self = std::move(*other);
}

QNoDebug* QNoDebug_Space(QNoDebug* self) {
	return new QNoDebug(self->space());
}

QNoDebug* QNoDebug_Nospace(QNoDebug* self) {
	return new QNoDebug(self->nospace());
}

QNoDebug* QNoDebug_MaybeSpace(QNoDebug* self) {
	return new QNoDebug(self->maybeSpace());
}

QNoDebug* QNoDebug_Quote(QNoDebug* self) {
	return new QNoDebug(self->quote());
}

QNoDebug* QNoDebug_Noquote(QNoDebug* self) {
	return new QNoDebug(self->noquote());
}

QNoDebug* QNoDebug_MaybeQuote(QNoDebug* self) {
	return new QNoDebug(self->maybeQuote());
}

QNoDebug* QNoDebug_Verbosity(QNoDebug* self, int param1) {
	return new QNoDebug(self->verbosity(param1));
}

QNoDebug* QNoDebug_MaybeQuote1(QNoDebug* self, const char param1) {
	return new QNoDebug(self->maybeQuote(param1));
}

void QNoDebug_Delete(QNoDebug* self) {
    delete self;
}

